porch/repository/pkg/oci/oci.go

```diff
@@ -165,6 +168,58 @@ func (r *ociRepository) buildPackageRevision(ctx context.Context, name ImageDige
 	return p, nil
 }
 
+func GetFunctionMeta(reference string, ctx context.Context) (*functionMeta, error) {
+	ref, err := name.ParseReference(reference)
+	if err != nil {
+		return nil, fmt.Errorf("parse image reference %v: %v", reference, err)
+	}
+	image, err := remote.Image(ref, remote.WithAuthFromKeychain(gcrane.Keychain), remote.WithContext(ctx))
+	if err != nil {
+		return nil, fmt.Errorf("pull remote image %v: %v", reference, err)
+	}
+	manifest, err := image.Manifest()
```

Comment:
we may want to explore caching of manifests (in a follow-up). the containerregistry library does some caching (or can be told to) but it doesn't cache some parts of the images. I forget the detail; something to follow-up on.

Reply:
Ah, that's a good idea! I'll explore the caching.